XenAdmin/Wizards/PatchingWizard/PatchingWizard_ModePage.cs

```diff
@@ -165,6 +185,29 @@ private void ManualRadioButton_CheckedChanged(object sender, EventArgs e)
             UpdateEnablement();
         }
 
+        private bool AnyPoolForbidsAutoRestart()
+        {
+            var poolForbidRestartKey = "XenCenter.CustomFields.hci-forbid-update-auto-restart";
+            foreach (var server in SelectedServers)
+            {
+                var pool = Helpers.GetPoolOfOne(server.Connection);
+                var poolOtherConfig = Helpers.GetOtherConfig(pool);
+                if (poolOtherConfig.ContainsKey(poolForbidRestartKey) &&
+                    poolOtherConfig[poolForbidRestartKey].ToLowerInvariant().Equals(bool.TrueString.ToLowerInvariant()))
+                {
+                    return true;
+                }
+            }
+
+            return false;
+        }
+
+        private bool PatchRequiresReboot()
+        {
+            return !textBoxLog.Text.Equals(Messages.PATCHINGWIZARD_MODEPAGE_NOACTION);

```

Comment:
Is there a better way to check this than comparing strings?

An approach I find better is to modify the ModeRetailPatch and ModeSuppPack methods to return a bool signifying whether action is required and adding an out string parameter to return the guidance text that will populate the box - then you check that bool.